### PR TITLE
Added folder and files for Macro-1 keyboard by Laneware Peripherals

### PR DESCRIPTION
**Pull request by Laneware**
## Description
Added folder and files for Macro-1 keyboard by Laneware Peripherals.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard Addition
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

(Ticked what I believe applys to adding a new keyboard, however this is my first time)

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
Please review the [PR Checklist](https://docs.qmk.fm/#/pr_checklist), and apply the changes indicated from there.

**Comment by Laneware** [comment]
> Please review the [PR Checklist](https://docs.qmk.fm/#/pr_checklist), and apply the changes indicated from there.

Changes have been made, thankyou :)

File: keyboards/macro1/config.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#pragma once
```
Comment (drashna) [inline keyboards/macro1/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/macro1/config.h [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4C50 // "LP" = Laneware Peripherals
+#define PRODUCT_ID      0x9999
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Laneware_Peripherals
+#define PRODUCT         Laneware_MACRO_1
+#define DESCRIPTION     Laneware_MACRO_1
```
Comment (drashna) [inline keyboards/macro1/config.h:11]:
```suggestion
```

File: keyboards/macro1/config.h [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
+
+/* prevent stuck modifiers */
+#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline keyboards/macro1/config.h:45]: These are deprecated.
```suggestion
```

File: keyboards/macro1/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,58 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/macro1/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/macro1/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (index == 0) { /* First encoder */
+        if (clockwise) {
+            tap_code(KC_VOLU);
+        } else {
+            tap_code(KC_VOLD);
+        }
+    }
+        return false; 
+}
```
Comment (drashna) [inline keyboards/macro1/keymaps/default/keymap.c:55]:
```suggestion
```

File: keyboards/macro1/keymaps/macropad/keymap.c [hunk 6]
```diff
@@ -0,0 +1,58 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/macro1/keymaps/macropad/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/macro1/keymaps/southpaw/keymap.c [hunk 7]
```diff
@@ -0,0 +1,58 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/macro1/keymaps/southpaw/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/macro1/keymaps/southpaw/keymap.c [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (index == 0) { /* First encoder */
+        if (clockwise) {
+            tap_code(KC_VOLU);
+        } else {
+            tap_code(KC_VOLD);
+        }
+    } 
+     return false; 
+}
```
Comment (drashna) [inline keyboards/macro1/keymaps/southpaw/keymap.c:55]:
```suggestion
```

File: keyboards/macro1/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,100 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/macro1/keymaps/via/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/macro1/keymaps/via/keymap.c [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+        if (clockwise) {
+            tap_code(KC_VOLU);
+        } else {
+            tap_code(KC_VOLD);
+        }
+    } 
+     return false; 
+}
+
+
```
Comment (drashna) [inline keyboards/macro1/keymaps/via/keymap.c:99]:
```suggestion
```

File: keyboards/macro1/macro1.h [hunk 11]
```diff
@@ -0,0 +1,99 @@
+#pragma once
```
Comment (drashna) [inline keyboards/macro1/macro1.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/macro1/macro1.c [hunk 12]
```diff
@@ -0,0 +1 @@
+#include "macro1.h"
```
Comment (drashna) [inline keyboards/macro1/macro1.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

```suggestion
#include "macro1.h"

bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) { /* First encoder */
        if (clockwise) {
            tap_code(KC_VOLU);
        } else {
            tap_code(KC_VOLD);
        }
    }
        return false; 
}
```

File: keyboards/macro1/readme.md [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+
+![Macro-1](https://geekhack.org/index.php?topic=111541.0)
+
+A versitile numpad/macropad (14x6) with a rotary encoder, made and sold by Laneware Peripherals.
+
+* Keyboard Maintainer: [Laneware Peripherals](https://github.com/laneware)
+* Hardware Supported: Macro-1
+* Hardware Availability: [Laneware Peripherals](https://lanewareperipherals.com/), 
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/macro1/readme.md:11]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/macro1/rules.mk [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+MOUSEKEY_ENABLE ?= yes	# Mouse keys(+4700)
+EXTRAKEY_ENABLE ?= yes	# Audio control and System control(+450)
+CONSOLE_ENABLE ?= no	# Console for debug(+400)
+COMMAND_ENABLE ?= no    # Commands for debug and configuration
+SLEEP_LED_ENABLE ?= no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE ?= yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE ?= no  # Enable keyboard backlight functionality
+AUDIO_ENABLE ?= no
+RGBLIGHT_ENABLE ?= no
+ENCODER_ENABLE = yes
```
Comment (drashna) [inline keyboards/macro1/rules.mk:20]:
```suggestion
BOOTMAGIC_ENABLE = yes    # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes     # Mouse keys
EXTRAKEY_ENABLE = yes     # Audio control and System control
CONSOLE_ENABLE = no       # Console for debug
COMMAND_ENABLE = no       # Commands for debug and configuration
SLEEP_LED_ENABLE = no     # Breathing sleep LED during USB suspend
NKRO_ENABLE = yes         # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
BACKLIGHT_ENABLE = no     # Enable keyboard backlight functionality
AUDIO_ENABLE = no
RGBLIGHT_ENABLE = no
ENCODER_ENABLE = yes
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/macro1/keymaps/default/keymap.c (hunk 5 above)
Comment (Laneware) [reply keyboards/macro1/keymaps/default/keymap.c:55]: Will this remove the encoder functionality?

Comment (Laneware) [reply keyboards/macro1/keymaps/default/keymap.c:55]: Never mind, I see that you have added it into the .c file

File: keyboards/macro1/config.h (hunk 1 above)
Comment (Laneware) [reply keyboards/macro1/config.h:1]: Change made

File: keyboards/macro1/config.h (hunk 2 above)
Comment (Laneware) [reply keyboards/macro1/config.h:11]: change made

File: keyboards/macro1/config.h (hunk 3 above)
Comment (Laneware) [reply keyboards/macro1/config.h:45]: changes made

File: keyboards/macro1/keymaps/default/keymap.c (hunk 4 above)
Comment (Laneware) [reply keyboards/macro1/keymaps/default/keymap.c:1]: change made

File: keyboards/macro1/keymaps/default/keymap.c (hunk 5 above)
Comment (Laneware) [reply keyboards/macro1/keymaps/default/keymap.c:55]: change made

File: keyboards/macro1/keymaps/macropad/keymap.c (hunk 6 above)
Comment (Laneware) [reply keyboards/macro1/keymaps/macropad/keymap.c:1]: change made

File: keyboards/macro1/keymaps/southpaw/keymap.c (hunk 7 above)
Comment (Laneware) [reply keyboards/macro1/keymaps/southpaw/keymap.c:1]: change made

File: keyboards/macro1/keymaps/southpaw/keymap.c (hunk 8 above)
Comment (Laneware) [reply keyboards/macro1/keymaps/southpaw/keymap.c:55]: change made

File: keyboards/macro1/keymaps/via/keymap.c (hunk 9 above)
Comment (Laneware) [reply keyboards/macro1/keymaps/via/keymap.c:1]: change made

File: keyboards/macro1/keymaps/via/keymap.c (hunk 10 above)
Comment (Laneware) [reply keyboards/macro1/keymaps/via/keymap.c:99]: change made

File: keyboards/macro1/macro1.h (hunk 11 above)
Comment (Laneware) [reply keyboards/macro1/macro1.h:1]: changes made

File: keyboards/macro1/macro1.c (hunk 12 above)
Comment (Laneware) [reply keyboards/macro1/macro1.c:1]: changes made

File: keyboards/macro1/readme.md (hunk 13 above)
Comment (Laneware) [reply keyboards/macro1/readme.md:11]: change made

File: keyboards/macro1/rules.mk (hunk 14 above)
Comment (Laneware) [reply keyboards/macro1/rules.mk:20]: change made

**Comment by Laneware** [comment]
I have updated the requested changes and pushed them to this pull request. I hope this is the way you wanted me to do it rather than through Git. I have requested re-review. Thanks for your help :)

**Review by drashna — APPROVED** [review]

File: keyboards/macro1/rules.mk [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+MOUSEKEY_ENABLE = yes     # Mouse keys
+EXTRAKEY_ENABLE = yes     # Audio control and System control
+CONSOLE_ENABLE = no       # Console for debug
+COMMAND_ENABLE = no       # Commands for debug and configuration
+SLEEP_LED_ENABLE = no     # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes         # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no     # Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = no
+ENCODER_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/macro1/rules.mk:20]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
```

File: keyboards/macro1/readme.md [hunk 16]
```diff
@@ -0,0 +1,22 @@
+# Macro-1
+
+![Macro-1](https://geekhack.org/index.php?topic=111541.0)
```
Comment (fauxpark) [inline keyboards/macro1/readme.md:3]: Needs to be a direct image link.

File: keyboards/macro1/info.json [hunk 17]
```diff
@@ -0,0 +1,84 @@
+{
+    "keyboard_name": "MACRO1", 
+    "url": "", 
+    "maintainer": "qmk", 
+    "layouts":{
+      "Numpad":{
```
Comment (fauxpark) [inline keyboards/macro1/info.json:6]:
```suggestion
      "LAYOUT_numpad":{
```

File: keyboards/macro1/info.json [hunk 18]
```diff
@@ -0,0 +1,84 @@
...
+          {"label":"1", "x":0, "y":4}, 
+          {"label":"2", "x":1, "y":4}, 
+          {"label":"3", "x":2, "y":4}, 
+          {"label":"Enter", "x":3, "y":4, "h":2}, 
+          {"label":"0", "x":0, "y":5, "w":2}, 
+          {"label":".", "x":2, "y":5}
+        ]
+      }, 
+
+      "Southpaw":{
```
Comment (fauxpark) [inline keyboards/macro1/info.json:31]:
```suggestion
      "LAYOUT_southpaw":{
```

File: keyboards/macro1/info.json [hunk 19]
```diff
@@ -0,0 +1,84 @@
...
+          {"label":"Enter", "x":0, "y":4, "h":2}, 
+          {"label":"1", "x":1, "y":4}, 
+          {"label":"2", "x":2, "y":4}, 
+          {"label":"3", "x":3, "y":4}, 
+          {"label":".", "x":1, "y":5}, 
+          {"label":"0", "x":2, "y":5, "w":2}
+        ]
+      },
+
+      "Macropad":{
```
Comment (fauxpark) [inline keyboards/macro1/info.json:56]:
```suggestion
      "LAYOUT_macropad":{
```

File: keyboards/macro1/config.h [hunk 20]
```diff
@@ -0,0 +1,55 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4C50 // "LP" = Laneware Peripherals
+#define PRODUCT_ID      0x9999
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Laneware_Peripherals
+#define PRODUCT         Laneware_MACRO_1
```
Comment (fauxpark) [inline keyboards/macro1/config.h:26]:
```suggestion
#define MANUFACTURER    Laneware Peripherals
#define PRODUCT         MACRO-1
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/macro1/rules.mk (hunk 15 above)
Comment (Laneware) [reply keyboards/macro1/rules.mk:20]: change made

File: keyboards/macro1/readme.md (hunk 16 above)
Comment (Laneware) [reply keyboards/macro1/readme.md:3]: change made

File: keyboards/macro1/info.json (hunk 17 above)
Comment (Laneware) [reply keyboards/macro1/info.json:6]: change made

File: keyboards/macro1/info.json (hunk 19 above)
Comment (Laneware) [reply keyboards/macro1/info.json:56]: change made

File: keyboards/macro1/info.json (hunk 18 above)
Comment (Laneware) [reply keyboards/macro1/info.json:31]: change made

File: keyboards/macro1/config.h (hunk 20 above)
Comment (Laneware) [reply keyboards/macro1/config.h:26]: Change made (can't use "-" in product definition)

Comment (fauxpark) [reply keyboards/macro1/config.h:26]: Yes you can: https://github.com/qmk/qmk_firmware/blob/master/keyboards/6ball/config.h#L28

Comment (Laneware) [reply keyboards/macro1/config.h]: thanks, updated.

**Comment by Laneware** [comment]
As per previous requested changes, I have updated the requested changes and pushed them to this pull request. I hope this is the way you wanted me to do it rather than through Git. I have requested re-review. Thanks for your help :)

File: keyboards/macro1/rules.mk [hunk 21]
```diff
@@ -0,0 +1,24 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+ENCODER_ENABLE = yes
+
```
Comment (fauxpark) [inline keyboards/macro1/rules.mk:9]:
```suggestion
```

File: keyboards/macro1/readme.md [hunk 22]
```diff
@@ -0,0 +1,22 @@
+# Macro-1
+
+![Macro-1](https://imgur.com/a/tV02eMK)
```
Comment (fauxpark) [inline keyboards/macro1/readme.md:3]:
```suggestion
![Macro-1](https://i.imgur.com/pnTgKuTh.jpeg)
```

File: keyboards/macro1/keymaps/southpaw/keymap.c [hunk 23]
```diff
@@ -0,0 +1,63 @@
...
+  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the 
+  * GNU General Public License for more details. 
+  * 
+  * You should have received a copy of the GNU General Public License 
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+  */ 
+
+#include QMK_KEYBOARD_H
+
+/* KEYMAP_southpaw
```
Comment (fauxpark) [inline keyboards/macro1/keymaps/southpaw/keymap.c:19]:
```suggestion
/*
```

File: keyboards/macro1/keymaps/via/keymap.c [hunk 24]
```diff
@@ -0,0 +1,103 @@
...
+  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the 
+  * GNU General Public License for more details. 
+  * 
+  * You should have received a copy of the GNU General Public License 
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+  */ 
+
+#include QMK_KEYBOARD_H
+
+/* KEYMAP_numpad
```
Comment (fauxpark) [inline keyboards/macro1/keymaps/via/keymap.c:19]:
```suggestion
/*
```

File: keyboards/macro1/keymaps/via/keymap.c [hunk 25]
```diff
@@ -0,0 +1,103 @@
...
+ * ├───┼───┼───┤   │ 
+ * │K30│K31│K32│   │
+ * ├───┼───┼───┼───┤ 
+ * │K40│K41│K42│K43│
+ * ├───┴───┼───┤   │ 
+ * │K50    │K52│   │
+ * └───────┴───┴───┘
+ */
+
+ /* KEYMAP_southpaw
```
Comment (fauxpark) [inline keyboards/macro1/keymaps/via/keymap.c:36]:
```suggestion
 /*
```

File: keyboards/macro1/keymaps/via/keymap.c [hunk 26]
```diff
@@ -0,0 +1,103 @@
...
+ * │   ├───┼───┼───┤ 
+ * │   │K31│K32│K33│
+ * ├───┼───┼───┼───┤ 
+ * │K40│K41│K42│K43│
+ * │   ├───┼───┴───┤ 
+ * │   │K51│K52    │
+ * └───┴───┴───────┘
+ */
+
+/* KEYMAP_macropad
```
Comment (fauxpark) [inline keyboards/macro1/keymaps/via/keymap.c:53]:
```suggestion
/*
```

File: keyboards/macro1/keymaps/macropad/keymap.c [hunk 27]
```diff
@@ -0,0 +1,63 @@
...
+  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the 
+  * GNU General Public License for more details. 
+  * 
+  * You should have received a copy of the GNU General Public License 
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+  */ 
+
+#include QMK_KEYBOARD_H
+
+/* KEYMAP_macropad
```
Comment (fauxpark) [inline keyboards/macro1/keymaps/macropad/keymap.c:19]:
```suggestion
/*
```

File: keyboards/macro1/keymaps/default/keymap.c [hunk 28]
```diff
@@ -0,0 +1,64 @@
...
+  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the 
+  * GNU General Public License for more details. 
+  * 
+  * You should have received a copy of the GNU General Public License 
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+  */ 
+
+#include QMK_KEYBOARD_H
+
+/* KEYMAP_numpad
```
Comment (fauxpark) [inline keyboards/macro1/keymaps/default/keymap.c:19]:
```suggestion
/*
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/macro1/rules.mk (hunk 21 above)
Comment (Laneware) [reply keyboards/macro1/rules.mk:9]: change made

File: keyboards/macro1/readme.md (hunk 22 above)
Comment (Laneware) [reply keyboards/macro1/readme.md:3]: change made

File: keyboards/macro1/keymaps/southpaw/keymap.c (hunk 23 above)
Comment (Laneware) [reply keyboards/macro1/keymaps/southpaw/keymap.c:19]: change made

File: keyboards/macro1/keymaps/via/keymap.c (hunk 24 above)
Comment (Laneware) [reply keyboards/macro1/keymaps/via/keymap.c:19]: change made

File: keyboards/macro1/keymaps/via/keymap.c (hunk 25 above)
Comment (Laneware) [reply keyboards/macro1/keymaps/via/keymap.c:36]: change made

File: keyboards/macro1/keymaps/macropad/keymap.c (hunk 27 above)
Comment (Laneware) [reply keyboards/macro1/keymaps/macropad/keymap.c:19]: change made

File: keyboards/macro1/keymaps/via/keymap.c (hunk 26 above)
Comment (Laneware) [reply keyboards/macro1/keymaps/via/keymap.c:53]: change made

File: keyboards/macro1/keymaps/default/keymap.c (hunk 28 above)
Comment (Laneware) [reply keyboards/macro1/keymaps/default/keymap.c:19]: change made

**Comment by Laneware** [comment]
Additional changes made, re-review requested, thanks :)

File: keyboards/macro1/rules.mk [hunk 29]
```diff
@@ -0,0 +1,23 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
```
Comment (fauxpark) [inline keyboards/macro1/rules.mk:23]:
```suggestion
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by Laneware** [comment]
Change made

File: keyboards/macro1/rules.mk (hunk 29 above)
Comment (Laneware) [reply keyboards/macro1/rules.mk:23]: change made

File: keyboards/macro1/rules.mk (hunk 21 above)
Comment (fauxpark) [reply keyboards/macro1/rules.mk:9]: This is still outstanding.

Comment (Laneware) [reply keyboards/macro1/rules.mk:9]: Can you please help me understand what you are after? I removed this and then you requested to add it back in.

Comment (fauxpark) [reply keyboards/macro1/rules.mk:9]: No, you removed the wrong one. Look at the file itself, this is in the wrong place.

Comment (Laneware) [reply keyboards/macro1/rules.mk]: I see now, sorry, fixed now.

File: keyboards/macro1/rules.mk [hunk 30]
```diff
@@ -0,0 +1,21 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+#   change yes to no to disable
```
Comment (fauxpark) [inline keyboards/macro1/rules.mk:7]:
```suggestion
# Build Options
#   change yes to no to disable
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/macro1/rules.mk (hunk 30 above)
Comment (Laneware) [reply keyboards/macro1/rules.mk:8]: added back in

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**